Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -524,4 +529,73 @@ public int getLATValue(Object adInfoObj) {
         }
         return LATVal_;
     }
+
+    /**
+     * <p>
+     * Method to prefetch the GAID and LAT values
+     * </p>
+     *
+     * @param callback {@link GAdsParamsFetchEvents} instance to notify process completion
+     * @return {@link Boolean} with true if GAID fetch process started.
+     */
+    public boolean prefetchGAdsParams(GAdsParamsFetchEvents callback) {
+        boolean isPrefetchStarted = false;
+        if (TextUtils.isEmpty(GAIDString_)) {
+            isPrefetchStarted = true;
+            new GAdsPrefetchTask(callback).execute();
+        }
+        return isPrefetchStarted;
+    }
+
+    /**
+     * <p>
+     * Async task to fetch GAID and LAT value
+     * This task fetch the GAID and LAT in background. The Background task times out
+     * After GAID_FETCH_TIME_OUT
+     * </p>
+     */
+    private class GAdsPrefetchTask extends AsyncTask<Void, Void, Void> {
```

Comment:
This whole asynctask feels excessive... Why don't you just put in the Thread above where you call `new GAdsPrefetchTask(callback).execute();`?